source/blog/044-rfc-indented-syntax.md

```diff
@@ -0,0 +1,121 @@
+---
+title: "Request for Comments: Indented Syntax Improvements"
+author: James Stuckey Weber
+date: 2024-11-27 15:30:00 -8
+---
+
+For users of the indented syntax (`.sass`), there has been a long standing
+request for the ability to have line breaks that don't end a statement. Certain
+CSS syntaxes like `grid-template` and `@font-face` `src` declarations are long,
+and line breaks can greatly increase the readability of the styles. 
+
+There have been many suggestions, several attempts, and numerous workarounds to
+address this pain point. We are proposing several new syntax options in the
+indented syntax that add considerable flexibility and look to address this as
+part of the language.
+
+## The new syntax
+
+Instead of adding syntax that is completely new to Sass, we propose to borrow
+syntax from the SCSS format and make it available in the indented syntax. This
+reduces the learning curve, code complexity, and likelihood of future conflicts
+with new CSS features.
+
+A common way of using this syntax will be wrapping lists with linebreaks in
+parentheses. This allows for a grid-template declaration on multiple lines.
+
+```sass
+.grid
+  display: grid
+  grid-template: (
+    "header" min-content
+    "main" 1fr
+  )
+}
+```
+
+These parentheses will not be included in the output.
+
+```css
+.grid {
+  display: grid;
+  grid-template: "header" min-content "main" 1fr;
+}
+```
+
+Powering this change is the underlying rule that linebreaks only end statements
+in places where statements can end. Anywhere where a statement can't end, a
+linebreak is treated as whitespace. This means that if the parser can't tell
+whether or not a linebreak is intended to end a statement, it will end the
+statement.
+
+This example demonstrates several places where line breaks will be possible.
+
+```sass
+@each $item in ¹
+    1 2 3
+  .item-#{ ²
+    $item
+  }
+    content: $item * ³
+        10
+```
+
+1. A statement can't end after the word "in" in an `@each` statement.
+2. A statement can't end inside the curly braces in an interpolation.
+3. A statement can't end after a multiplication operator.
```

Comment:
I think these references might be easier to parse if they used line numbers instead of superscript. Or I'm not sure if there's a better way to make it clear that they're not part of the code example?

Reply:
Line numbers made it harder to follow, I think.

I'll push a version with inline comments-
<img width="829" alt="image" src="https://github.com/user-attachments/assets/2a291082-2fa2-4e2d-83ae-d0377f852943">

Here's the original-
<img width="589" alt="image" src="https://github.com/user-attachments/assets/50444dd3-8807-4cd2-99a5-02ca1c4da045">